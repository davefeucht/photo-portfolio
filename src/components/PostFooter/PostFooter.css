.post-footer {
  background-color: #000000;
  font-size: min(3vw, 1rem);
  color: #FFFFFF;
  grid-area: footer;
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  transition: width var(--modal-transition-time), height var(--modal-transition-time)
}

.post-footer .labels {
  margin: auto auto auto 5px;
}