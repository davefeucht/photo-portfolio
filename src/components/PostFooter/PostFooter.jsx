/****************
* PostTitlebar component displays the titlebar for a Post
****************/

import React from "react";
import { observer } from 'mobx-react';
import CloseButton from '../CloseButton/CloseButton.jsx';
import './PostFooter.css';

const PostFooter = observer(({ tags }) => {
  return(
    <div className="post-footer">
      <div className="labels">{tags.join(', ')}</div>
    </div>
  )
});

PostFooter.displayName = 'PostFooter';

export default PostFooter;