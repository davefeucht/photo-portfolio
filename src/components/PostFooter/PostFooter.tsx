/** **************
* PostFooter component displays the footer for a Post
*************** */

import './PostFooter.css';

import { observer } from 'mobx-react';
import * as React from 'react';

import { Store } from '../../utils/types';

interface PostFooterProps {
    stateStore: Store
}

const PostFooter: React.FC<PostFooterProps> = ({ stateStore }) => {
    return (
        <div className="post-footer">
            <div className="labels">{stateStore.visiblePost.tagNames.join(', ')}</div>
        </div>
    );
};

PostFooter.displayName = 'PostFooter';

export default observer(PostFooter);
