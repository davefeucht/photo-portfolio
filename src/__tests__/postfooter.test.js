import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import PostFooter from '../components/PostFooter/PostFooter';
import stateStore from '../StateStore/store';

test('PostFooter displays', () => {
  const store = new stateStore();
  let component;
  let tree;
  component = renderer.create(
      <MemoryRouter >
          <PostFooter stateStore={store}></PostFooter>
      </MemoryRouter>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
