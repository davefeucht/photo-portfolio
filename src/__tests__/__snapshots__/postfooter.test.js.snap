// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFooter displays 1`] = `
<div
  className="post-footer"
>
  <div
    className="labels"
  />
</div>
`;
